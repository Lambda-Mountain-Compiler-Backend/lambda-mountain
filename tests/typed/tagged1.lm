
type A Tag1
     | (Tag2( S ))
     | (Tag3( S S ));

(Tag2 3);
