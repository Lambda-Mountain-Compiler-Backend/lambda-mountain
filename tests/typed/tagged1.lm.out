((() (Type ((Literal A) (App ((App ((App ((App ((Literal Tag1) (Literal |))) (App ((Literal Tag2) (Literal S))))) (Literal |))) (App ((Literal Tag3) (App ((Literal S) (Literal S)))))))))) (GExpr (: ((App ((: ((Literal Tag2) (And ((-> (S (Tag2 S))) Atom)))) (: ((Literal 3) Atom)))) (Tag2 S)))))
